src/boot/efi/boot.c

```diff
@@ -767,44 +784,15 @@ static BOOLEAN menu_run(
 
                 case KEYPRESS(0, 0, '-'):
                 case KEYPRESS(0, 0, 'T'):
-                        if (config->timeout_sec_efivar > 0) {
+                        if (config->timeout_sec_efivar != TIMEOUT_UNSET)
                                 config->timeout_sec_efivar--;
```

Comment:
so if the value is currently 1 this will turn off the menu. which is ok. but if the menu is already off, then this will cause the menu to be always shown. which i think is also ok (though i'd prefer if that would be code explicitly, rather than through silent overflow. But then things become ugly, because then it becomes unset where finally saturation is reached.

I think it makes sense to saturate, and i think it makes sense to go from 2s → 1s → no menu → menu without timeout. But then we should stay there, and i think the steps should be explicitly coded, just to make this easier to read and follow